src/main/java/org/elasticsearch/cluster/ClusterName.java

```diff
@@ -23,18 +23,47 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Streamable;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
 
 /**
  *
  */
-public class ClusterName implements Streamable {
+public class ClusterName extends AbstractClusterStatePart implements Streamable {
+
+    public static final String TYPE = "cluster_name";
```

Comment:
I think in-general we should try to keep all the `TYPE` constants in a single place. If at all possible, it might not be due to extendibility, I'd really prefer an Enum here, we can still just use the name of the enum internally